statsmodels/graphics/functional.py

```diff
@@ -1,14 +1,441 @@
+    its contours.
+
+    Finally, using these contours, the different quantiles can be extracted
+    allong with the median curve and the outliers.
+
+    References
+    ----------
+    [1] R.J. Hyndman and H.L. Shang, "Rainbow Plots, Bagplots, and Boxplots for
+        Functional Data", vol. 19, pp. 29-45, 2010.
+
+    Examples
+    --------
+    Load the El Nino dataset.  Consists of 60 years worth of Pacific Ocean sea
+    surface temperature data.
+
+    >>> import matplotlib.pyplot as plt
+    >>> import statsmodels.api as sm
+    >>> data = sm.datasets.elnino.load()
+
+    Create a functional boxplot.  We see that the years 1982-83 and 1997-98 are
+    outliers; these are the years where El Nino (a climate pattern
+    characterized by warming up of the sea surface and higher air pressures)
+    occurred with unusual intensity.
+
+    >>> fig = plt.figure()
+    >>> ax = fig.add_subplot(111)
+    >>> res = sm.graphics.hdrboxplot(data.raw_data[:, 1:],
+    ...                              labels=data.raw_data[:, 0].astype(int),
+    ...                              ax=ax)
+
+    >>> ax.set_xlabel("Month of the year")
+    >>> ax.set_ylabel("Sea surface temperature (C)")
+    >>> ax.set_xticks(np.arange(13, step=3) - 1)
+    >>> ax.set_xticklabels(["", "Mar", "Jun", "Sep", "Dec"])
+    >>> ax.set_xlim([-0.2, 11.2])
+
+    >>> plt.show()
+
+    .. plot:: plots/graphics_functional_hdrboxplot.py
+
+    """
+    fig, ax = utils.create_mpl_ax(ax)
+
+    data = np.asarray(data)
+    if xdata is None:
+        xdata = np.arange(data.shape[1])
+
+    n_samples, dim = data.shape
+    # PCA and bivariate plot
+    pca = PCA(data, ncomp=ncomp)
+    data_r = pca.factors
+
+    # Create gaussian kernel
+    ks_gaussian = KDEMultivariate(data_r, bw=bw,
+                                  var_type='c' * data_r.shape[1])
+
+    # Boundaries of the n-variate space
+    bounds = np.array([data_r.min(axis=0), data_r.max(axis=0)]).T
+
+    # Compute contour line of pvalue linked to a given probability level
+    if alpha is None:
+        alpha = [threshold, 0.9, 0.5]
+    else:
+        alpha.extend([threshold, 0.9, 0.5])
+        alpha = list(set(alpha))
+    alpha.sort(reverse=True)
+
+    n_quantiles = len(alpha)
+    pdf_r = ks_gaussian.pdf(data_r).flatten()
+    pvalues = [np.percentile(pdf_r, (1 - alpha[i]) * 100,
+                             interpolation='linear')
+               for i in range(n_quantiles)]
+
+    # Find mean, outliers curves
+    if have_de_optim:
+        median = differential_evolution(lambda x: - ks_gaussian.pdf(x),
+                                        bounds=bounds, maxiter=5).x
+    else:
+        median = brute(lambda x: - ks_gaussian.pdf(x),
+                       ranges=bounds, finish=fmin)
+
+    outliers = np.where(pdf_r < pvalues[alpha.index(threshold)])[0]
+    if labels is not None:
+        labels = list(itertools.compress(labels, outliers))
+    outliers = data[outliers]
```

Comment:
I think the outliers index should be returned also.
AFAICS, currently users get the outlier arrays but no label or index for which series are outliers.

Reply:
Agreed, it will now return the indices. Maybe this is more convenient that returning the labels as the user can be interested in finding other data using the indices.